### PR TITLE
fix(callAPIMethod): bug ff & safari

### PR DESCRIPTION
**Pull request by ya-woodcutter**
ff & safari (isomorphic-fetch) doesn't support response.body

### CODE REVIEW

File: src/callAPIMethod.js [hunk 1]
```diff
@@ -26,13 +26,16 @@ export default function callAPI(
 
     return fetch(url, {...defaults.fetchOptions, ...fetchOptions, ...options})
         .then( result => {
-            const promiseCallback = result.body ? result[method]() : new Promise(resolve => resolve(result.body));
+            const promiseCallback = result[method]();
```
Comment (TimeRaider) [inline src/callAPIMethod.js]: @ya-woodcutter according to https://developer.mozilla.org/en-US/docs/Web/API/Response
Response implements Body. I think the issue is with isomorphic-fetch not being really isomorphic and using node-fetch underneath. 
Did you try using Response.blob() instead?
```js
const promiseCallback =  result[result.body ? method : 'blob']();
```

Comment (ya-woodcutter) [inline src/callAPIMethod.js]: >  I think the issue is with isomorphic-fetch not being really isomorphic...

Isomorphic-fetch implements the minimal functionality of fetch.

> Did you try using Response.blob() instead?

I didn't try to do it. But I think that the transform data from Blob to the right data type will be problematic. It requires additional conditions. 

I'll try to do something in this way.

**Comment by ya-woodcutter** [comment]
Now:
-
- We are trying to call `response[method]()`
- **First catch:** 
  - if `!ok` then `throw new Error(status, statusText)`
  - if `ok` then return `body || null` (If we are in this `catch`, then an error in getting data)
- **Then:**
  - if `!ok` then `throw new Error(status, statusText, body)`
  - if `ok` then return `body`
- **Second catch:** return error

**Comment by TimeRaider** [comment]
@ya-woodcutter Sorry I've fucked up your pr

**Comment by ya-woodcutter** [comment]
@TimeRaider ? emm... it's ok :)

**Comment by ya-woodcutter** [comment]
UP: @TimeRaider @evilj0e @Blackheart340

File: src/callAPIMethod.js [hunk 2]
```diff
@@ -33,17 +33,20 @@ export default function callAPI(
...
-            }
-
-            return responseBodyPromise
-                .then( responseBody => {
-                    throw new APIError(response.status, response.statusText, responseBody);
-                });
-        })
+        .then( response => response[method]()
+                .catch( () => {
+                    if (!response.ok) {
```
Comment (Blackheart340) [inline src/callAPIMethod.js]: Let's keep the same [codestyle](https://github.com/auru/unity-api/blob/master/src/callAPIMethod.js#L32) everywhere

**Pull request merged**